iib/web/models.py

```diff
@@ -1013,6 +1054,17 @@ def from_json(cls, kwargs, batch=None):
 
         request = cls(**request_kwargs)
         request.add_state('in_progress', 'The request was initiated')
+        db.session.add(request)
```

Comment:
I would prefer having this logic in the `api_v1.yaml` for all the requests.

Reply:
Looks like similar code to me for all requests. Maybe you can create a helper method in ` api_v1.yaml` ?

Reply:
I'm not sure I understand your request. Logic in yaml?

Reply:
Sorry I meant api_v1.py

Reply:
So for all the three types of requests, let's have this piece of code in `api_v1.py` file instead.